edge-functions/alerts/alert_types/shard_failed.ts

```diff
@@ -0,0 +1,71 @@
+import { AlertRecord, EmailConfig } from "../index.ts";
+import { commonTemplate, getTaskDetailsPageURL } from "../template.ts";
+import { Recipient } from "../template.ts";
+import { ControllerAlertArguments } from "./controller_alerts.ts";
+
+type ShardFailedRecord = AlertRecord<"shard_failed", ControllerAlertArguments>;
+
+const formatAlertEmail = ({
+  arguments: { recipients, spec_type },
+  catalog_name,
+}: ShardFailedRecord): EmailConfig[] => {
+  const subject = `Estuary Flow: Shard failure detected for ${spec_type} ${catalog_name}`;
```

Comment:
nit: Remove "Shard" and just call it "Failure detected" ? Seems like we can avoid introducing the "shard" terminology here